juju/juju
# Fix for LP 1777897 - openstack cert add-cloud and bootstrap

## Description of change

Allow the OpenStack client to use a ca cert for communication with the cloud.
This does not apply to the cinder client and ca cert use with the juju
storage provider functionality.

The cloud config variable "ca-certificates" takes a list of strings.

Change the juju add-cloud OpenStack url verification to use a client
which does not do ssl hostname validation.  This allows the url to be
validated even if a ca cert is required.

Drive by change of strings to constants in openstack provider config.go

## QA steps

Using an openstack deployment requiring a CACert for access:

1. juju add-cloud, the endpoint url should validate.
2. Add the ca cert to the clouds.yaml for the new cloud:
clouds:
  openstack:
    type: openstack
.....
    ca-certificates:
      - |
        -----BEGIN CERTIFICATE-----
       .....
        -----END CERTIFICATE-----
2. juju bootstrap <openstack>

There should be no change with an openstack deployment not requiring a CACert.

## Documentation changes

N/A

## Bug reference

https://bugs.launchpad.net/juju/+bug/1777897
